routes/project.js

```diff
@@ -434,11 +434,30 @@ module.exports = function (express) {
     });
   });
 
-  router.delete('/:projectId', util.isLoggedIn, function (req, res) {
+  router.delete('/:projectId', util.forbidAnonymousUser, function (req, res) {
     var projectId = req.params.projectId;
     var user = req.user;
 
     async.waterfall([
+      // Check if project was created by current user.
+      function (callback) {
+        Project.find({'_id': projectId}, function (err, projects) {
+          if (err) {
+            return callback(err);
+          }
+
+          if (projects.length === 0) {
+            return callback('Not found project');
+          }
+
+          var project = projects[0];
+          if (project.user.toString() !== user._id.toString()) {
+            return callback('Not user project');
```

Comment:
What about having the same info for `Not found project` and for `Not user project`. 
For example `Invalid project`
This should be a bit safer as the attacker cannot confirm that project id is correct.

Reply:
Ok, fixed.